tools/runfiles/runfiles.rs

```diff
@@ -69,15 +80,31 @@ impl Runfiles {
 fn find_runfiles_dir() -> io::Result<PathBuf> {
     let exec_path = std::env::args().nth(0).expect("arg 0 was not set");
 
-    let mut binary_path = PathBuf::from(&exec_path);
+    let mut binary_path = PathBuf::from(&exec_path)
+        .canonicalize()
```

Comment:
Why the call to `canonicalize`? This will plausibly break consumers who use symlinks

Reply:
Oh yeah... This fails when running tests. I'll remove it to demonstrate the error.

Reply:
Or just find an old run: https://buildkite.com/bazel/rules-rust-rustlang/builds/2145#0f14a372-551e-439d-8513-33c1875a894f

Because the path of `binary_path` is relative, we don't traverse far enough to find the `runfiles` directory.

Reply:
Would love some suggestions on how to better handle this 🙏

Reply:
We can absolutise; something like:
```rust
if !binary_path.is_absolute() {
  binary_path = std::env::current_dir().expect("Failed to get current directory").join(binary_path);
}
```

But we should definitely not resolve symlinks.

Reply:
That seemed to do the trick 😄